agent/auto-config/config_translate.go

```diff
@@ -57,19 +58,21 @@ import (
 // maintenance burden. In the long run we should unify the protobuf Config and
 // the normal agent/config.Config so that we can just serialize the protobuf version
 // without any translation. For now, this hack is necessary :(
-func translateConfig(c *pbconfig.Config) map[string]interface{} {
-	out := map[string]interface{}{
-		"datacenter":         c.Datacenter,
-		"primary_datacenter": c.PrimaryDatacenter,
-		"node_name":          c.NodeName,
+func translateConfig(c *pbconfig.Config) config.Config {
+	result := config.Config{
+		Datacenter:        stringPtr(c.Datacenter),
+		PrimaryDatacenter: stringPtr(c.PrimaryDatacenter),
+		NodeName:          stringPtr(c.NodeName),
 	}
 
 	// only output the SegmentName in the configuration if its non-empty
 	// this will avoid a warning later when parsing the persisted configuration
 	if c.SegmentName != "" {
-		out["segment"] = c.SegmentName
+		result.SegmentName = stringPtr(c.SegmentName)
 	}
 
+	/* TODO:
```

Comment:
If this approach looks viable I can complete this function.

Reply:
Originally this function did return a `Config` but that was causing issues with the marshalling/unmarshalling process. When marshalled the Config would have a bunch of outer structs set including enterprise specific configuration and then when loaded it was emitting warnings.

Without the need to go to a JSON string most of the original function can probably just be put back.

Reply:
Oh, awesome! I found it in git, I will restore it and use it as the starting point